lib/bolt/inventory/inventory2.rb

```diff
@@ -17,12 +17,19 @@ def initialize(data, config = nil, plugins: nil, target_vars: {}, target_facts:
         # Config is saved to add config options to targets
         @config = config || Bolt::Config.default
         @data = data ||= {}
-        @groups = Group2.new(data.merge('name' => 'all'))
+        @data = Bolt::Util.walk_vals(data) do |value|
+          if value.is_a?(Hash) && value.include?('_plugin')
+            # How do we reliably instantiate new plugin object by name?
```

Comment:
I realize the way it has been implemented in this WIP is not correct. Would like to discuss how to do this.